servicetest/devicenode/test_devicenode_gateway.py

```diff
@@ -0,0 +1,194 @@
+        "name": "/dev/tty0",
+        "mode": 644
+    }
+]
+
+""" Following capabilities are fed to DynamicConfiguration tests to examine the case
+    when new mode is configured for same device.
+"""
+test_cap_0 = Capability("double.activation.same.device.0.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_0}
+                        ])
+
+test_cap_1 = Capability("double.activation.same.device.1.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_1}
+                        ])
+
+""" Following capabilities are fed to DynamicConfiguration tests to examine the case
+    when a new device is added to configuration
+"""
+test_cap_2 = Capability("double.activation.new.device.0.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_0}
+                        ])
+
+test_cap_3 = Capability("double.activation.new.device.1.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_NEW_DEV_1}
+                        ])
+
+""" Following capability is fed to DynamicConfiguration tests to examine the case
+    when exactly same device with same mode is added
+"""
+test_cap_4 = Capability("double.activation.same.device.same.mode.cap",
+                        [
+                            {"id": "devicenode", "config": GW_CONFIG_SAME_DEV_SAME_MODE}
+                        ])
+
+""" Define the manifest so it can be used by the testframework  """
+manifest = StandardManifest(TESTOUTPUT_DIR,
+                            "devicenode-test-manifest.json",
+                            [
+                                test_cap_0,
+                                test_cap_1,
+                                test_cap_2,
+                                test_cap_3,
+                                test_cap_4
+                            ])
+
+
+def service_manifests():
+    """ The agent fixture calls this function when it creates the service manifests
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [manifest]
+
+# Test suites
+
+
+@pytest.mark.usefixtures("testhelper", "agent")
+class TestDeviceNode(object):
+    """ This suite tests the behavior of DeviceNodeGateway within given set of capabilities.
+        It uses the DeviceNodeGateway to assert behavior.
+
+        The tests use the Agent D-Bus interface to enable gateway configurations,
+        for a specified list of capabilities.
+    """
+
+    """ Followinng decorator presents parametrization of argument list to test function.
+        each argument consist two capabilitties to applied on gateway. The test is responsible
+        for appliying each of them sequentially.
+
+        To see differences and aims of parameters please see comments on declaration
+    """
```

Comment:
Ok, I see. Yes the reference to the definition of the caps is enough I guess, should be clear enough. I think it's enough to reference that and skip the first paragraph of this comment, it only states what a parametrization decorator is and does, and that should be known by the reader already.